web/modules/custom/custom_migration_users/src/Plugin/migrate/destination/MyCustomFields.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+namespace Drupal\custom_migration_users\Plugin\migrate\destination;
+
+use Drupal\migrate\Annotation\MigrateDestination;
+use Drupal\migrate\Plugin\MigrateDestinationInterface;
+use Drupal\migrate\Row;
+
+/**
+ * Custom migrate destination plugin for users extra fields.
+ *
+ * @MigrateDestination(
+ *   id = "my_custom_fields"
+ * )
+ */
+class MyCustomFields implements MigrateDestinationInterface {
```

Comment:
There is a base class for destination, you don't need to implement interface and all of it's methods directly unless you are doing something very specific